### PR TITLE
gh-890: Use git switch to switch to branch

### PR DESCRIPTION
**Pull request by za**
instead of using git checkout

https://github.com/python/devguide/issues/890

### CODE REVIEW

**Comment by terryjreedy** [comment]
To me, a non-expert get user, this creates confusion.  Is there any difference?  Any advantage either way?  Need both alternatives be given every time?

**Comment by za** [comment]
@terryjreedy There's no difference. `git switch` and `git restore` was introduced in git 2.23. I think, we don't need both alternatives given every time.

**Comment by encukou** [comment]
Right. IMO it would be best to remove `git checkout` from the bootcamp entirely. New users don't need to learn about it at all.

Background: `git checkout` does two very different things depending on how you call it (switching branches and restoring changed files). Git developers replaced it by two separate commands (and took the opportunity to also change all the options, so they make more sense). They're keeping `checkout` around for existing scripts and finger memories.

File: gitbootcamp.rst [hunk 1]
```diff
@@ -112,7 +112,10 @@ This is equivalent to::
 
    # create a new branch from main, without checking it out
    git branch <branch-name> main
+
    # check out the branch
+   git switch <branch-name>
+   # or using old command
    git checkout <branch-name>
```
Comment (ezio-melotti) [inline gitbootcamp.rst:119]:
```suggestion
   # create a new branch from main
   git branch <branch-name> main
   # switch to the new branch
   git switch <branch-name>
```

File: gitbootcamp.rst [hunk 2]
```diff
@@ -128,6 +131,8 @@ To list all the branches, including the remote branches::
 
 To switch to a different branch::
 
+   git switch <another-branch-name>
+   # or using old command
    git checkout <another-branch-name>
```
Comment (ezio-melotti) [inline gitbootcamp.rst:136]:
```suggestion
```

File: pullrequest.rst [hunk 3]
```diff
@@ -110,7 +110,11 @@ You should have already :ref:`set up your system <setup>`,
   to your branch, making more changes, committing them, and pushing them to
   automatically update your PR::
 
+   # switch to your branch
+   git switch <branch-name>
+   # or using old command
    git checkout <branch-name>
+
```
Comment (ezio-melotti) [inline pullrequest.rst:117]:
```suggestion
   git switch <branch-name>
```

File: pullrequest.rst [hunk 4]
```diff
@@ -129,7 +133,10 @@ You should have already :ref:`set up your system <setup>`,
   will show a warning to this end and you may be asked to address this. Merge
   the changes from the main branch while resolving the conflicts locally::
 
+   git switch <branch-name>
+   # or using old command
    git checkout <branch-name>
+
```
Comment (ezio-melotti) [inline pullrequest.rst:139]:
```suggestion
```

File: setup.rst [hunk 5]
```diff
@@ -97,7 +97,7 @@ in the ``cpython`` directory and two remotes that refer to your own GitHub fork
 If you want a working copy of an already-released version of Python,
 i.e., a version in :ref:`maintenance mode <maintbranch>`, you can checkout
 a release branch. For instance, to checkout a working copy of Python 3.8,
-do ``git checkout 3.8``.
+do ``git switch 3.8`` or using the old comand ``git checkout 3.8``.
```
Comment (ezio-melotti) [inline setup.rst:100]:
```suggestion
do ``git switch 3.8``.
```

**Review by ezio-melotti — CHANGES_REQUESTED** [review]

**Review by ezio-melotti — APPROVED** [review]

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**